src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateDirectionsQuestionnaireCallbackHandler.java

```diff
@@ -28,10 +28,19 @@
 public class GenerateDirectionsQuestionnaireCallbackHandler extends CallbackHandler {
 
     private static final List<CaseEvent> EVENTS = Collections.singletonList(GENERATE_DIRECTIONS_QUESTIONNAIRE);
+    private static final String TASK_ID = "DefendantResponseFullDefenceGenerateDirectionsQuestionnaire";
+    private static final String CLAIMANT_TASK_ID = "ClaimantResponseGenerateDirectionsQuestionnaire";
 
     private final DirectionsQuestionnaireGenerator directionsQuestionnaireGenerator;
     private final ObjectMapper objectMapper;
 
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return callbackParams.getCaseData().getApplicant1ResponseDate() != null
+            ? CLAIMANT_TASK_ID
+            : TASK_ID;
```

Comment:
As above, I don't think we need that logic, this event is triggered by separate business processes.